templates/Cargo.nix.tera

```diff
@@ -27,7 +27,16 @@ args@{
 }:
 let
   nixifiedLockHash = "{{ cargo_lock_hash }}";
-  workspaceSrc = if args.workspaceSrc == null then ./. else args.workspaceSrc;
+  workspaceSrc = lib.fileset.toSource rec {
+    root = if args.workspaceSrc == null then ./. else args.workspaceSrc;
```

Comment:
Filtering of the source should be done at the user interface, when using the overlay.  If it's done in the Cargo.nix, it will be inflexible.